src/solve.jl

```diff
@@ -298,8 +298,8 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.
 
   if typeof(alg) <: OrdinaryDiffEqCompositeAlgorithm
     id = CompositeInterpolationData(f,timeseries,ts,ks,alg_choice,dense,cache)
-    beta2 === nothing && ( beta2=beta2_default(alg.algs[cache.current]) )
-    beta1 === nothing && ( beta1=beta1_default(alg.algs[cache.current],beta2) )
+    beta2 === nothing && ( beta2=_composite_beta2_default(alg.algs, cache.current, QT) )
+    beta1 === nothing && ( beta1=_composite_beta1_default(alg.algs, cache.current, QT) )
```

Comment:
Would it be sufficient to use
```suggestion
    beta2 === nothing && ( beta2=QT(beta2_default(alg.algs[cache.current])) )
    beta1 === nothing && ( beta1=QT(beta1_default(alg.algs[cache.current],beta2)) )
```
for type inference of `beta1` and `beta2`?

Reply:
No because the type of `alg.algs[cache.current]` is not known in advance because `alg.algs` is a tuple of different element types. So I did the whole unrolling thing to make sure Julia assigns `alg.algs[1]` to a different variable from `alg.algs[2]`. Your suggestion creates a type instability in the intermediate variable that `alg.algs[cache.current]` is assigned to.

Reply:
But that shouldn't matter due to the function barrier, should it? I just saw that you called the QT constructor in the generated function but that could be done without generated function as well.

Reply:
it shouldn't matter that much but it's still a dynamic dispatch